content/docs/tasks/traffic-management/custom-gateway/index.md

```diff
@@ -0,0 +1,372 @@
+                    - s390x
+              preferredDuringSchedulingIgnoredDuringExecution:
+              - weight: 2
+                preference:
+                  matchExpressions:
+                  - key: beta.kubernetes.io/arch
+                    operator: In
+                    values:
+                    - amd64
+              - weight: 2
+                preference:
+                  matchExpressions:
+                  - key: beta.kubernetes.io/arch
+                    operator: In
+                    values:
+                    - ppc64le
+              - weight: 2
+                preference:
+                  matchExpressions:
+                  - key: beta.kubernetes.io/arch
+                    operator: In
+                    values:
+                    - s390x
+            podAntiAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+              - labelSelector:
+                  matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                    - istio-internal-custom
+                topologyKey: kubernetes.io/hostname
+
+    {{< /text >}}
+
+1. Create your service:s
+
+    {{< warning_icon >}} The `NodePort` used needs to be an available Port.
+
+    {{< text yaml >}}
+
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: istio-internal-custom
+      annotations:
+        service.beta.kubernetes.io/aws-load-balancer-type: nlb
+      labels:
+        app: istio-internal-custom
+        istio: istio-internal-custom
+    spec:
+      type: LoadBalancer
+      selector:
+        app: istio-internal-custom
+        istio: istio-internal-custom
+      ports:
+        -
+          name: http2
+          nodePort: 32380
+          port: 80
+          targetPort: 80
+        -
+          name: https
+          nodePort: 32390
+          port: 443
+        -
+          name: tcp
+          nodePort: 32400
+          port: 31400
+
+    {{< /text >}}
+
+1. Create your Istio custom gateway configuration object:
+
+    {{< text yaml >}}
+
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      annotations:
+      name: istio-custom-gateway
+      namespace: default
+    spec:
+      selector:
+        istio: istio-internal-custom
+      servers:
+      - hosts:
+        - '*.mydemo.com'
+        port:
+          name: http
+          number: 80
+          protocol: HTTP
+        tls:
+          httpsRedirect: true
+      - hosts:
+        - '*.mydemo.com'
+        port:
+          name: https
+          number: 443
+          protocol: HTTPS
+        tls:
+          mode: SIMPLE
+          privateKey: /etc/istio/ingressgateway-certs/tls.key
+          serverCertificate: /etc/istio/ingressgateway-certs/tls.crt
+
+    {{< /text >}}
+
+**Congratulations!** You can now use your custom `istio-custom-gateway` [gateway](/docs/reference/config/istio.networking.v1alpha3/#Gateway) configuration object.
```

Comment:
What's missing is an actual running service that you can access via mydemo.com to see it works. That's needed if this is going to be an example.

Reply:
The purpose of my initial post was to explain how to configure a custom gateway with example, to understand how it's works

Reply:
Even if you want to explain details, I still think users should be told to generate with helm, instead of copy and paste deployments, etc., from the example doc. Something more like this: https://deploy-preview-1984--preliminary-istio.netlify.com/docs/examples/advanced-gateways/wildcard-https-egress-gateway/#custom-egress-gateway-with-sni-proxy

Reply:
So how does user confirm that it works? That's the usual last step of an Istio example.

Reply:
I agree with @frankbu regarding an actual running service. I would also clarify in the beginning how this task is different from https://istio.io/docs/tasks/traffic-management/secure-ingress/ and what is the motivation.

Something like: The Securing Gateways with HTTPS task describes how to use configure ingress HTTPS traffic by using the Istio ingress gateway. This task describes how to deploy a custom ingress gateway, using cert-manager.